runtime/lua/vim/treesitter/query.lua

```diff
@@ -737,87 +790,50 @@ end
 --- Lists the currently available predicates to use in queries.
 ---@return string[] : Supported predicates.
 function M.list_predicates()
-  return vim.tbl_keys(predicate_handlers)
-end
-
-local function xor(x, y)
-  return (x or y) and not (x and y)
-end
-
-local function is_directive(name)
-  return string.sub(name, -1) == '!'
+  return vim.tbl_keys(guard_handlers)
 end
 
 ---@private
+---@param pattern_i integer
+---@param guards ProcessedGuard[]
 ---@param match TSQueryMatch
 ---@param source integer|string
-function Query:match_preds(match, source)
-  local _, pattern = match:info()
-  local preds = self.info.patterns[pattern]
-
-  if not preds then
-    return true
-  end
-
-  local captures = match:captures()
-
-  for _, pred in pairs(preds) do
-    -- Here we only want to return if a predicate DOES NOT match, and
```

Comment:
Also removed this comment, as unknown predicates throw an error since https://github.com/neovim/neovim/pull/12826

Reply:
Range selection seems to be broken. This is the second time I selected one range and it displays as a different range.